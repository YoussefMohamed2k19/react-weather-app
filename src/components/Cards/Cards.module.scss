.itemCard {
  border: 2px solid #dadada;
  border-radius: 4px;
  height: 130px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.itemCard:nth-child(1) img {
  width: 65px;
  /* border: 2px solid red; */
  margin-top: -12px;
}

.itemCard:nth-child(1) {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}
