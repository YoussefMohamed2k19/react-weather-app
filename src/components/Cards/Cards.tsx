import styles from "./Cards.module.scss";

interface ICards {
  weather?: object[];
  data: JSX.Element | string | number;
  label: string;
  unit?: string;
  icon?: string;
}

const Cards = ({ weather, data, label, unit, icon }: ICards) => {
  return (
    <div className={styles.itemCard}>
      {weather ? (
        <img
          src={`http://openweathermap.org/img/wn/${weather
            ?.filter((x: any, y: number) => y === 0)
            ?.map((x: any) => x.icon)}@2x.png`}
          alt=""
        />
      ) : (
        <i className={`fs-2 ${icon}`}></i>
      )}
      <span className="fs-6 fw-bold">
        {data} {unit}
      </span>
      <span className="fw-lighter text-secondary">{label}</span>
    </div>
  );
};

export default Cards;
