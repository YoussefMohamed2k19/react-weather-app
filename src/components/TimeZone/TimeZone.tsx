import dayjs from "dayjs";
import utc from "dayjs/plugin/utc";
dayjs.extend(utc);

interface ITimeZone {
  timezone: number;
  input: number;
}

const TimeZone = ({ timezone, input }: ITimeZone) => {
  return (
    <>
      {dayjs
        .unix(input)
        .utc()
        .utcOffset(timezone / 3600)
        .format("h : mm A")}
    </>
  );
};

export default TimeZone;
