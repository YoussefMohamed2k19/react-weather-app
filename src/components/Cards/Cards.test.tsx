import { render, screen } from "@testing-library/react";
import Cards from "./Cards";
import TimeZone from "../TimeZone/TimeZone";

describe("Check Card Component", () => {
  const data = 15.88;
  const label = "temperature";
  const unit = "Â° C";
  const icon = "bi bi-droplet";
  const weather = [
    { id: 804, main: "Clouds", description: "overcast clouds", icon: "04d" },
  ];

  test("Checks if a card with icon and data loads or not", () => {
    const { container } = render(
      <Cards icon={icon} label={label} unit={unit} data={data} />
    );
    expect(container).toMatchSnapshot();
  });

  test("Checks if a card with image and data loads or not", () => {
    const { container } = render(
      <Cards weather={weather} data={data} label={label} unit={unit} />
    );
    expect(container).toMatchSnapshot();
  });

  test("Checks if a card with local time and icon loads or not", () => {
    const input = 1662523476;
    const timezone = 7200;
    const { container } = render(
      <Cards
        data={<TimeZone input={input} timezone={timezone} />}
        label="Sun Rise"
        icon="bi bi-sunrise"
      />
    );
    expect(container).toMatchSnapshot();
  });
});
