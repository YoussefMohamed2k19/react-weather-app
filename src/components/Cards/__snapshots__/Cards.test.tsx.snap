// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Card Component Checks if a card with icon and data loads or not 1`] = `
<div>
  <div
    class="itemCard"
  >
    <i
      class="fs-2 bi bi-droplet"
    />
    <span
      class="fs-6 fw-bold"
    >
      15.88
       
      ° C
    </span>
    <span
      class="fw-lighter text-secondary"
    >
      temperature
    </span>
  </div>
</div>
`;

exports[`Check Card Component Checks if a card with image and data loads or not 1`] = `
<div>
  <div
    class="itemCard"
  >
    <img
      alt=""
      src="http://openweathermap.org/img/wn/04d@2x.png"
    />
    <span
      class="fs-6 fw-bold"
    >
      15.88
       
      ° C
    </span>
    <span
      class="fw-lighter text-secondary"
    >
      temperature
    </span>
  </div>
</div>
`;

exports[`Check Card Component Checks if a card with local time and icon loads or not 1`] = `
<div>
  <div
    class="itemCard"
  >
    <i
      class="fs-2 bi bi-sunrise"
    />
    <span
      class="fs-6 fw-bold"
    >
      6 : 04 AM
       
    </span>
    <span
      class="fw-lighter text-secondary"
    >
      Sun Rise
    </span>
  </div>
</div>
`;
